earn/src/components/portfolio/modal/WithdrawModal.tsx

```diff
@@ -81,21 +82,39 @@ function WithdrawButton(props: WithdrawButtonProps) {
     address: kitty.address,
     abi: KittyABI,
     functionName: 'convertToShares',
-    args: [ethers.utils.parseUnits(withdrawAmount || '0.00', token.decimals).toString()],
+    args: [withdrawAmount.toFixed(0)],
     chainId: activeChain.id,
-  });
+  }) as { data: BigNumber | undefined; isLoading: boolean };
+
+  const bigRequestedShares = requestedShares ? toBig(requestedShares) : new Big(0);
+  // Being extra careful here to make sure we don't withdraw more than the user has
+  const finalWithdrawAmount = bigRequestedShares.gt(maxRedeemBalance) ? maxRedeemBalance : bigRequestedShares;
```

Comment:
Since we're technically redeeming by calling the `redeem` function (not withdrawing), let's call this something like `numberOfSharesToRedeem`